### ISSUE TITLE
Visualization problems even markers are correctly detected

### ISSUE THREAD

**Issue by Baptiste-AIST** [meyerls/aruco-estimator#17]
Hello, I'm having some visualization issues, I checked if the aruco marker is correctly detected in the views and it seems correct.
But when I use the visualization it gives something wrong, I'm not sure how we can debug that

![example_1](https://user-images.githubusercontent.com/105469463/193165060-cc59266d-3da2-47e5-a4b2-87f2d694151e.jpg)
![example_2](https://user-images.githubusercontent.com/105469463/193165127-45f2da74-9c1f-45a2-8181-35d4f66fe183.jpg)

It might not be very clear on the pictures, but the marker position is missed. I don't know where the problem comes from and how we can investigate that.

Also concerning the scaling factor, I'm wondering about the size of the marker, you put in cm (like 15 for 15cm), but when I do so it gives an incorrect scaling factor, but if I put in meters (0.15 for 15cm) then it starts giving me a more accurate result.
I got confused because in your `download.py` file you wrote `self.scale = 0.15  # cm` is it 0.15cm or 0.15m? I think meter makes more sense. 

Thank you

**Comment by meyerls** [comment]
Okay, do you mind sending me your photos? Then I can check in detail what is going wrong.

As for the size of the Aruco marker, it is correct that it should be 0.15m. I'll change the default settings.

**Comment by meyerls** [comment]
I finally found the reason for the faulty analysis and visualization. It seems that I only used the model of the simple pinhole camera for my purposes. Since the camera you chose is a radial camera, I mistakenly assigned a negative value (k1 - radial distortion coefficient) to the principal point in the y-direction. As a result, all calculations were incorrect. 

![Screenshot from 2022-10-01 13-46-35](https://user-images.githubusercontent.com/23578706/193408078-8b334a3f-2f3d-4504-b399-6702c5dcb309.png)

![Screenshot from 2022-10-01 13-45-57](https://user-images.githubusercontent.com/23578706/193408081-c7827c7c-8614-408e-afc9-a4f7c23c8bdf.png)

Additionally I might suggest to undistort your images (by creating a dense model) to further increase the accuracy of the scaling process. 

@Baptiste-AIST  thanks a lot for the provided data!
